performance/tests/test_eqprs_uploads.py

```diff
@@ -153,3 +166,89 @@ def test_with_correct_csv(self):
         assert indicator.annual_dpme_coordinator_comments == ""
         assert indicator.annual_treasury_comments == ""
         assert indicator.annual_audited_output == ""
+
+    def test_task_creation(self):
+        fy = FinancialYear.objects.create(slug="2021-22")
+        sphere = Sphere.objects.create(name="Provincial", financial_year=fy)
+        government = Government.objects.create(name="Eastern Cape", sphere=sphere)
+        department = Department.objects.create(
+            name="Health", government=government, vote_number=1
+        )
+
+        model_admin = EQPRSFileUploadAdmin(
+            model=EQPRSFileUpload, admin_site=AdminSite()
+        )
+        model_admin.save_model(
+            obj=EQPRSFileUpload(file=self.csv_file),
+            request=Mock(user=self.superuser),
+            form=None,
+            change=None,
+        )
+
+        last_element = EQPRSFileUpload.objects.all().last()
+        assert last_element.task_id is not None
+
+    def test_status_in_list_view(self):
+        assert "processing_completed" in EQPRSFileUploadAdmin.list_display
+
+        fy = FinancialYear.objects.create(slug="2021-22")
+        sphere = Sphere.objects.create(name="Provincial", financial_year=fy)
+        government = Government.objects.create(name="Eastern Cape", sphere=sphere)
+        department = Department.objects.create(
+            name="Health", government=government, vote_number=1
+        )
+
+        model_admin = EQPRSFileUploadAdmin(
+            model=EQPRSFileUpload, admin_site=AdminSite()
+        )
+        model_admin.save_model(
+            obj=EQPRSFileUpload(file=self.csv_file),
+            request=Mock(user=self.superuser),
+            form=None,
+            change=None,
+        )
+
+        last_element = EQPRSFileUpload.objects.all().last()
+        assert model_admin.processing_completed(last_element) == True
+
+    def test_with_national_government(self):
+        fy = FinancialYear.objects.create(slug="2021-22")
+        sphere = Sphere.objects.create(name="Provincial", financial_year=fy)
+        government = Government.objects.create(name="South Africa", sphere=sphere)
+        department = Department.objects.create(
+            name="Health", government=government, vote_number=1
+        )
+
+        test_element = EQPRSFileUpload.objects.create(
+            user=self.superuser, file=self.national_file
+        )
+        performance.admin.save_imported_indicators(test_element.id)
+        test_element.refresh_from_db()
+
+        assert test_element.import_report == ""
+        assert test_element.num_imported == 1
+        indicator = models.Indicator.objects.all().first()
+        assert indicator.programme_name == "Programme 1: Administration"
+
+    def test_with_alias(self):
```

Comment:
please also add test coverage for

- if gov A and dept A is in the data, its old indicators for that financial year are deleted. Any other financial years remain
- if Gov A, B and dept A, B is in the data, old indicators for Gov A DeptA are deleted but Gov A dept B are not deleted

Reply:
done 👍